server/core/src/main/java/io/whitefox/core/types/predicates/LeafEvaluationResult.java

```diff
@@ -0,0 +1,29 @@
+package io.whitefox.core.types.predicates;
+
+import io.whitefox.core.ColumnRange;
+import io.whitefox.core.types.DataType;
+import java.util.Optional;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class LeafEvaluationResult {
+
+  Optional<Pair<ColumnRange, String>> rangeEvaluationResult;
+  Optional<Pair<Pair<DataType, String>, Pair<DataType, String>>> partitionEvaluationResult;
+
```

Comment:
Can we create a dedicated value class instead of using Pair? In order to improve readability